lib/ts/recipe/emailverification/components/themes/emailVerification/sendVerifyEmail.tsx

```diff
@@ -86,7 +88,11 @@ export const EmailVerificationSendVerifyEmail: React.FC<SendVerifyEmailThemeProp
         [props.config, props.recipeImplementation, props.onEmailAlreadyVerified]
     );
 
-    const handleSendError = useCallback(() => {
+    const handleSendError = useCallback(async () => {
+        // If the error cleared the session we should redirect to auth
+        if ((await Session.getInstanceOrThrow().doesSessionExist({ userContext })) !== true) {
+            await props.redirectToAuth();
+        }
```

Comment:
Add a TODO stating that when we add SessionAuth with requireAuth=true, we should remove this.